samples/cpp/tutorial_code/ImgTrans/HoughLines_Demo.cpp

```diff
@@ -53,22 +58,28 @@ int main( int argc, char** argv )
    /// Pass the image to gray
    cvtColor( src, src_gray, COLOR_RGB2GRAY );
 
-   /// Apply Canny edge detector
-   Canny( src_gray, edges, 50, 200, 3 );
+   /// Apply Canny/Sobel edge detector
+   Canny( src_gray, canny_edge, 50, 200, 3 );
+   Sobel( src_gray, sobel_edge, CV_16S, 1, 0 ); // dx(order of the derivative x)=1,dy=0
 
    /// Create Trackbars for Thresholds
    char thresh_label[50];
    snprintf( thresh_label, sizeof(thresh_label), "Thres: %d + input", min_threshold );
-
+   char edge_thresh_label[50];
+   sprintf( edge_thresh_label, "Edge Thres: input" );
```

Comment:
No need to use sprintf here.

Reply:
I want to add Edge threshold slider as this figure (
![image](https://user-images.githubusercontent.com/13181036/208776678-94512916-044d-41b4-b8aa-81c52d06ddb9.png)
), because weighted Hough depends on edge amount.